### ISSUE TITLE
Upgrade to use Go 1.8

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#1491]
I think it is time to align with a later Go version. 1.9 is out now. In the documentation as well as in the CI systems we align with 1.7. It is time to update. 

Doing so would not also give the opportunity to make some build improvements since 1.9 handles the vendor directory a bit differently now.

**Comment by praveenkumar** [comment]
There is only one problem, currently from CentOS/RHEL side we don't have 1.9 available yet but 1.8 is there quite some time and we should consider that for this iteration till 1.9 get listed in official repository of RHEL/CentOS. 

@hferentschik If that sound good then I will take this issue and try to update all the CI systems.

**Comment by hferentschik** [comment]
To gain some additional benefits in relation to the build environment we need 1.9. I am using 1.8 for quite some time and it works just fine, but it "just" offers a Go version update (including potential Go bug fixes). With 1.9 we can do a bit more. 

We can as a stepping stone go to 1.8 first, but we should then create a new issues for that. This would really just affect the CI configs and the docs. We can then at a later point in time make a jump to 1.9. 

Personally, I think not having a CentOS/RHEL package for Go 1.9 should not be a show stopper. After all you can install it manually - https://golang.org/dl/. In fact this is probably the better solution anyways. This way you can install multiple versions of Go in parallel and switch between them by changing a symbolic link.

**Comment by praveenkumar** [comment]
> We can as a stepping stone go to 1.8 first, but we should then create a new issues for that. This would really just affect the CI configs and the docs. We can then at a later point in time make a jump to 1.9.

Yes agree to that.

>Personally, I think not having a CentOS/RHEL package for Go 1.9 should not be a show stopper. After all you can install it manually - https://golang.org/dl/. In fact this is probably the better solution anyways. This way you can install multiple versions of Go in parallel and switch between them by changing a symbolic link.

@hferentschik I do agree but on downstream side to build our CDK binary, we can only able to use what provided in the official repository so I still think to have same golang version to build and test on upstream side.

**Comment by gbraad** [comment]
>  After all you can install it manually - https://golang.org/dl/. In fact this is probably the better solution anyways

I would prefer to stick with packaged versions, as that will make deployment a lot easier and consistent. These also live in cache on the CentOS CI side.

> We can as a stepping stone go to 1.8 first,

This sounds to me as the better approach... for now.

**Comment by hferentschik** [comment]
@praveenkumar, fair enough. As a stepping stone we can go to 1.8 first. Let's do this in another issue though.

**Comment by praveenkumar** [comment]
@hferentschik sure I will modify this issue text and create an new issue for follow up.

**Comment by gbraad** [comment]
Actually at first slightly confused, as a new issue #1511 was created with the same title. But it seems you modified this to reflect 1.8 first?

**Comment by praveenkumar** [comment]
> But it seems you modified this to reflect 1.8 first?

Yes, that's right so in this iteration, we will upgrade to 1.8 and then 1.9

**Comment by praveenkumar** [comment]
Resolved by #1516
